python/ray/serve/_private/deployment_state.py

```diff
@@ -1368,34 +1401,22 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
-
-        app_msg = f" in application '{self.app_name}'" if self.app_name else ""
-        logger.info(
-            f"Deploying new version of deployment {self.deployment_name}{app_msg}."
-        )
-
-    def _set_target_state_autoscaling(self, num_replicas: int) -> None:
-        """Update the target number of replicas based on an autoscaling decision.
-
-        This differs from _set_target_state because we are updating the
-        target number of replicas base on an autoscaling decision and
-        not a redeployment. This only changes the target num_replicas,
-        and doesn't change the current deployment status.
-        """
-
-        new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(num_replicas)
-        new_info.version = self._target_state.version.code_version
-
-        target_state = DeploymentTargetState.from_deployment_info(new_info)
+        # Determine whether the updated target state is simply upscaling/downscaling
+        is_scale = target_state.compare_state_excluding_num_replicas(self._target_state)
+        if self._target_state.info and is_scale:
+            new, old = (target_state.num_replicas, self._target_state.num_replicas)
+            scaling_decision = "Upscaling" if new > old else "Downscaling"
```

Comment:
Also, we should probably use the `DeploymentStatus` eum directly here instead of the string "Upscaling" and "Downscaling"

Reply:
New and old shouldn't be equal because for both manual scaling (changing `num_replicas`) and autoscaling, there are defensive checks that return early if nothing has changed. I'll add an assertion here.